Stripe/PublicHeaders/STPShippingAddressViewController.h

```diff
@@ -0,0 +1,92 @@
+//  Stripe
+//
+//  Created by Ben Guo on 8/29/16.
+//  Copyright © 2016 Stripe, Inc. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+#import <PassKit/PassKit.h>
+#import "STPTheme.h"
+#import "STPPaymentContext.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol STPShippingAddressViewControllerDelegate;
+
+/** This view controller contains a shipping address collection form. It renders a right bar button item that submits the form, so it must be shown inside a `UINavigationController`. Depending on your configuration's shippingType, the view controller may present a shipping method selection form after the user enters an address.
+ */
+@interface STPShippingAddressViewController : UIViewController
+
+/**
+ *  A convenience initializer; equivalent to calling `initWithConfiguration:[STPPaymentConfiguration sharedConfiguration] theme:[STPTheme defaultTheme] currency:nil shippingAddress:nil selectedShippingMethod:nil`.
+ */
+- (instancetype)init;
+
+/**
+ *  Initializes a new `STPShippingAddressCardViewController` with the provided parameters.
+ *
+ *  @param configuration             The configuration to use (this determines the required shipping address fields and shipping type). @see STPPaymentConfiguration
+ *  @param theme                     The theme to use to inform the view controller's visual appearance. @see STPTheme
+ *  @param currency                  The currency to use when displaying amounts for shipping methods. The default is USD.
+ *  @param shippingAddress           If set, the shipping address view controller will be pre-filled with this address. @see STPAddress
+ *  @param selectedShippingMethod    If set, the shipping methods view controller will use this method as the selected shipping method. If `selectedShippingMethod` is nil, the first shipping method in the array of methods returned by your delegate will be selected.
+ */
+- (instancetype)initWithConfiguration:(STPPaymentConfiguration *)configuration
+                                theme:(STPTheme *)theme
+                             currency:(nullable NSString *)currency
+                      shippingAddress:(nullable STPAddress *)shippingAddress
+               selectedShippingMethod:(nullable PKShippingMethod *)selectedShippingMethod;
+
+/**
+ *  The view controller's delegate. This must be set before showing the view controller in order for it to work properly. @see STPShippingAddressViewControllerDelegate
+ */
+@property(nonatomic, weak) id<STPShippingAddressViewControllerDelegate> delegate;
+
+/**
+ *  If you're pushing `STPShippingAddressViewController` onto an existing `UINavigationController`'s stack, you should use this method to dismiss it, since it may have pushed an additional shipping method view controller onto the navigation controller's stack.
+ *
+ *  @param hostViewController the view controller that the shipping address view controller was pushed onto.
+ */
+- (void)dismissWithHostViewController:(UIViewController *)hostViewController;
```

Comment:
Ah yeah, good call. I've changed this to `dismissWithCompletion` in 86144ae , and added the same method to `STPPaymentMethodsViewController` (but not `STPAddCardViewController`, since that VC doesn't push any view controllers on itself, and can just be dismissed normally).